common_types/src/record_batch.rs

```diff
@@ -269,19 +269,30 @@ impl RecordBatch {
     }
 }
 
-impl TryFrom<ArrowRecordBatch> for RecordBatch {
-    type Error = Error;
+#[derive(Default)]
+pub struct RecordBatchBuilder {
+    schema_cache: Option<RecordSchema>,
+}
 
-    fn try_from(arrow_record_batch: ArrowRecordBatch) -> Result<Self> {
-        let record_schema =
-            RecordSchema::try_from(arrow_record_batch.schema()).context(ConvertArrowSchema)?;
+impl RecordBatchBuilder {
+    pub fn build(&mut self, arrow_record_batch: ArrowRecordBatch) -> Result<RecordBatch> {
+        let schema = match &self.schema_cache {
+            Some(schema) => schema.clone(),
```

Comment:
Is it possible that the schema is not consistent?

Reply:
Now,  such usecase not exist.
But it is needed to be considered actually, I will add some comments about it.

Reply:
Some comments about are added.